keras_nlp/layers/random_insertion.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from keras import backend
+from tensorflow import keras
+import random
+
+class RandomInsertion(keras.layers.Layer):
+    """Augments input by randomly inserting words.
+
+    Args:
+        rate: A float in [0, 1] that is the rate of insertion
+        max_replacements: An integer that is the maximum number of insertions
+        insertion_list: A list of strings that are the words to insert
+        insertion_fn: fn that takes in a token and returns a insertion token.
+        insertion_py_fn: A python function that takes in a token and returns a
+            insertion token.
+        skip_list: A list of words to skip.
+        skip_fn: A function that takes a word and returns True if the word
+            should be skipped. This must be a traceable function of tf
+            operations.
+        skip_py_fn: A function that takes a word and returns True if the words
+            should be skipped. Unlike skip_fn, this can be any python function
+            that operates on strings, and does not need to use tf operations.
+        seed: A seed for the rng.
+
+
+    Examples:
+
+    Word Level usage
+    >>> def replace_word(word):
```

Comment:
I am feeling confused about the usage here. This layer is called RandomInsertion, but here it gets a `replace_word` fn, which looks like a use case for RandomReplace?